Sources/MessagingPush/Type/PushNotification.swift

```diff
@@ -3,9 +3,115 @@ import Foundation
 // A data type that represents a push notification.
 // Note: This data class represents *all* push notifications, even those not sent by CIO.
 public protocol PushNotification {
+    // an id that uniquely identifies this push notification.
+    // For push notifications sent by APN or FCM, this id gets populated by them with a random value.
+    // For locally created notifications, this ID is provided by the app developer. So, the ID might be a hard-coded value and not unique between 2 pushes.
     var pushId: String { get }
-    var deliveryDate: Date { get }
-    var title: String { get }
-    var message: String { get }
-    var data: [AnyHashable: Any] { get }
+
+    // Date when the push was delivered to the device.
+    // Some subclasses of UserNotifications framework does not include a delivery date so we have to keep this optional.
+    // Depending on the request that the framework makes to our SDK will determine if we get a date or not.
+    var deliveryDate: Date? { get }
```

Comment:
good call out. Are we using it, if so what for, and if not, why are we adding it?

Reply:
> Are we using it, if so what for

Yes, [it's used to make sure we do not handle the same push 2+ times](https://github.com/customerio/customerio-ios/blob/2f8dddfbc20a24ae1c56abf5d6ed501106f4201a/Sources/MessagingPush/Store/PushHistory.swift#L20).